File: python/stack_sort.py
class StackSort:

    @staticmethod
    def sort(a):
        b = []

        while a:
            x = a.pop()
            while b and b[-1] < x:
                a.append(b.pop())
            b.append(x)

        while b:
            a.append(b.pop())


if __name__ == "__main__":
    import random
    random.seed(0)

    stack_a = []
    for i in range(32):
      stack_a.append(random.randint(0,100))

    print(stack_a)
    StackSort.sort(stack_a)
    print(stack_a)

File: java/src/main/java/freezingoctosansa/StackSort.java
package freezingoctosansa;

import java.util.Stack;

public class StackSort {

  public static void sort(Stack<Integer> stackA) {

    Stack<Integer> stackB = new Stack<Integer>();

    while (!stackA.empty()) {

      Integer a = stackA.pop();

      while (!stackB.isEmpty() && stackB.peek() < a) {
        stackA.push(stackB.pop());
      }

      stackB.push(a);
    }

    while(!stackB.empty()) stackA.push(stackB.pop());
  }
}
